app/components/Icons/CollectionIcon.tsx

```diff
@@ -16,42 +17,49 @@ type Props = {
   size?: number;
   /** The color of the icon, defaults to the collection color */
   color?: string;
+  className?: string;
 };
 
 function ResolvedCollectionIcon({
   collection,
   color: inputColor,
   expanded,
   size,
+  className,
 }: Props) {
   const { ui } = useStores();
 
-  // If the chosen icon color is very dark then we invert it in dark mode
-  // otherwise it will be impossible to see against the dark background.
-  const color =
-    inputColor ||
-    (ui.resolvedTheme === "dark" && collection.color !== "currentColor"
-      ? getLuminance(collection.color) > 0.09
-        ? collection.color
-        : "currentColor"
-      : collection.color);
+  if (!collection.icon || collection.icon === "collection") {
```

Comment:
We need the `negation` check since collections whose icon is `collection` have `null` set in the DB currently.